catboost/libs/metrics/optimal_const_for_loss.h

```diff
@@ -29,6 +29,32 @@ namespace NCB {
         return targetSum / summaryWeight;
     }
 
+    inline float CalculateWeightedTargetAverageWithMissingValues(TConstArrayRef<float> target, TConstArrayRef<float> weights) {
+        TVector<float> targetWithoutNans;
```

Comment:
`CalculateWeightedTargetAverage` is very simple function and it will be ok if you just copy it here with small modifications for NaNs. It will be more short and without target and weights copying.

Reply:
Ok. It makes sense also to me. 
Done